### PR TITLE
[FIX][connector] switch unlink of producer into the new api to avoid old...

### PR DESCRIPTION
**Pull request by JonathanNEMRY**
.../new odoo confusion
- Create a Database
- Install "Products & Pricelists"  & "Purchase" & "Sales Management"
- Settings/Configuration/Purchases --> Manage different units of measure for products
- Settings/Configuration/Purchases --> Manage pricelist per supplier
- Invoicing/Create customer "customer"
- Édit "customer" and change its "Purchase Pricelist" (create a second one before)
- Install connector module
- Edit the created "customer" and change its "Purchase Pricelist" with the default one (it must be the real default purchase pricelist because it will make an unlink)
- Save and see "TypeError: unlink() takes at least 4 arguments (1 given)"
## Into 

```
`@api.model
 def set_multi(self, name, model, values):`
```

of `ir.property`
## Behavior without connector:

the `prop.unlink()` is well redirected to 
    `BaseModel.unlink(cr, uid, ids, context)` passing by the wrapper/new_api

**--> works pretty well**

---
## Behavior with connector:

the `prop.unlink()` is NOT well redirected:

```
`producer.unlink(cr, uid, ids, context)` passing by the wrapper/OLD_API
```

**--> takes at least 4 arguments (1 given)**
## Supposion:

I personally do not think this is a `connector` bug because it mostly appears as a 
`old/new api` problem (seems that `unlink` is called as new `api` from `ir.property`
but not defined as such into `BaseModel`).

**BUT** it can be resolved by switching the `producer.unlink` into new `api`.

If you have some other solution and/or any ideas

(Note that test will fail:  maybe rebase after merging this https://github.com/OCA/connector/pull/31)

-jne

### CODE REVIEW

File: connector/producer.py [hunk 1]
```diff
@@ -73,13 +73,13 @@ def write(self, vals):
...
+def unlink(self):
     if self.pool.get('connector.installed') is not None:
-        if not hasattr(ids, '__iter__'):
-            ids = [ids]
-        session = ConnectorSession(cr, uid, context=context)
+        session = ConnectorSession(self.env.cr, self.env.uid,
+                                   context=self.env.context)
         if on_record_unlink.has_consumer_for(session, self._name):
-            for record_id in ids:
+            for record_id in self._ids:
```
Comment (guewen) [inline connector/producer.py]: could you use `self.ids`?

**Comment by guewen** [comment]
Thanks! Seems fine, just one little thing with the `self.ids`.

**Comment by lmignon** [comment]
:+1:  once the the self._ids will be fixed :smile:

**Comment by AdilHoumadi** [comment]
:+1:

**Comment by pedrobaeza** [comment]
:+1:

**Comment by ghost** [comment]
:+1:

**Comment by guewen** [comment]
Just waiting after travis...

**Comment by guewen** [comment]
:+1:

**Comment by pedrobaeza** [comment]
UNIT_TEST is provoking this slowness...

**Comment by guewen** [comment]
I think that we have a record of thumbs up per hour in the OCA on this PR.

**Comment by pedrobaeza** [comment]
:laughing:

**Comment by JonathanNEMRY** [comment]
Rebase after merge of #31

**Comment by guewen** [comment]
I don't think the rebase was necessary because travis run the test with the branch merged on the HEAD of the base branch.

**Comment by JonathanNEMRY** [comment]
@guewen  thanks for this information

**Comment by guewen** [comment]
Should be green after merge of https://github.com/OCA/connector/pull/34
Anyway the failure is not related so I proceed with the merge.

Thanks!

**Comment by lepistone** [comment]
Hi all,

after some investigation I found out what was going on here.

The original `unlink` method is not decorated, and written with the old API. This is OK, because the `api.Meta` metaclass is used in the `BaseModel` and guesses the decorator automatically when it is not provided. In that case, the connector replaces the `unlink` method with a new, non decorated one, and the guessing never takes place (because the method is not defined in `BaseModel`).

This PR fixed the problem because the decorator `@api.multi` has been specified. I think just adding the old-style `@api.cr_uid_ids_context` would have worked as well.

Thanks to you all for your time!

**Pull request merged**